rxandroidble/src/test/groovy/com/polidea/rxandroidble/RxBleClientTest.groovy

```diff
@@ -315,4 +313,31 @@ class RxBleClientTest extends Specification {
         mock.getAddress() >> scanData['deviceMac']
         bleAdapterWrapperSpy.addScanResult(mock, scanData['rssi'], scanData['scanRecord'])
     }
+
+    def "should not call stopLeScan if startLeScan wasn't called before"() {
```

Comment:
But it's something different I guess. Calling `stopLeScan` and then `startLeScan` is not bad I guess. The bad thing is calling `stopLeScan` after a call `startLeScan` but before `startLeScan` returns.